test/tools/jsontests/LegacyTestsBoost.cpp

```diff
@@ -0,0 +1,221 @@
+#include <test/tools/jsontests/BlockChainTests.h>
+#include <test/tools/jsontests/StateTests.h>
+#include <test/tools/libtesteth/TestHelper.h>
+
+#include <boost/test/unit_test.hpp>
+
+using namespace dev::test;
+namespace fs = boost::filesystem;
+
+fs::path StateTestSuiteLegacyConstantinople::suiteFolder() const
+{
+    return "LegacyTests/Constantinople/GeneralStateTests";
+}
+
+fs::path StateTestSuiteLegacyConstantinople::suiteFillerFolder() const
+{
+    return "LegacyTests/Constantinople/GeneralStateTestsFiller";
+}
+
+fs::path BCGeneralStateTestsSuiteLegacyConstantinople::suiteFolder() const
+{
+    return "LegacyTests/Constantinople/BlockchainTests/GeneralStateTests";
+}
+
+fs::path BCGeneralStateTestsSuiteLegacyConstantinople::suiteFillerFolder() const
+{
+    return "LegacyTests/Constantinople/BlockchainTestsFiller/GeneralStateTests";
+}
+
+template <class T>
+class LegacyTestFixtureBase
+{
+public:
+    LegacyTestFixtureBase(bool _onlyRunIfOptionsAllSet)
+    {
+        T suite;
+        if (Options::get().fillchain || Options::get().filltests)
+            BOOST_FAIL("Legacy tests are sealed and not refillable!");
+
+        string casename = boost::unit_test::framework::current_test_case().p_name;
+        boost::filesystem::path suiteFillerPath = suite.getFullPathFiller(casename).parent_path();
+        static vector<string> const timeConsumingTestSuites{
+            string{"stTimeConsuming"}, string{"stQuadraticComplexityTest"}};
+
+        bool skipTheTest = false;
+        if (test::inArray(timeConsumingTestSuites, casename) && !test::Options::get().all)
+            skipTheTest = true;
+        if (_onlyRunIfOptionsAllSet && !Options::get().all)
```

Comment:
I suggest to simplify this by calling the parameter flag just `_skip` and checking `Options::get().all` on the calling side
(that is you'll have `LegacyConstantinopleGeneralStateTestFixture() : LegacyTestFixtureBase(false) {}` and `LegacyConstantinopleBCGeneralStateTestFixture() : LegacyTestFixtureBase(!test::Options::get().all) {}`

Reply:
I would not say it simplify. Sometimes it better be readable then optimized. 
I would replace bool with Enum